include/swift/SIL/TypeLowering.h

```diff
@@ -96,35 +96,120 @@ enum IsDependent_t : unsigned {
   IsNotDependent = false,
   IsDependent = true
 };
-  
+
+/// Is a lowered SIL type trivial?  That is, are copies ultimately more
+/// than just bit-copies, and it does it take any work to destroy?
+enum IsTrivial_t : bool {
+  IsNotTrivial = false,
+  IsTrivial = true
+};
+
+/// Is a lowered SIL type fixed-ABI?  That is, can the current context
+/// assign it a fixed size and alignment and perform value operations on it
+/// (such as copies, destroys, constructions, and projections) without
+/// metadata?
+///
+/// Note that a type can be frozen (non-resilient) without being fixed-ABI
```

Comment:
We haven't quite settled on the use of "frozen" for structs yet, so it's probably better not to use this. It's also possible to have a non-frozen type that is nonetheless fixed-ABI: an `@objc` enum.